watchtower/wtclient/session_negotiator.go

```diff
@@ -358,8 +350,15 @@ func (n *sessionNegotiator) createSession(tower *wtdb.Tower,
 		sessionKeyDesc, n.cfg.SecretKeyRing,
 	)
 
-	for _, lnAddr := range tower.LNAddrs() {
-		err := n.tryAddress(sessionKey, keyIndex, tower, lnAddr)
+	addr := tower.Addresses.Peek(true)
```

Comment:
When reading this code, I have no idea what the boolean parameter is supposed to do, so that I have to look it up. In this case I'd love to see "PeekAndLock" and "Peek" as public functions, so that the existing Peek(boolean) function can be made private (assuming go supports this concept).